lib/screens/sign_in.dart

```diff
@@ -105,14 +107,25 @@ class _SignInScreenState extends State<SignInScreen> {
                         RaisedButton(
                           key: Key('sign_in_submit_btn'),
                           color: Theme.of(context).primaryColor,
-                          onPressed: () {
+                          onPressed: () async {
                             if (_formKey.currentState.validate()) {
                               setState(() {
                                 _waitingForAuthResult = true;
                               });
                             } else {
                               _autoValidateForm = true;
                             }
+
+                            final authResult = await context
+                                .read<AuthService>()
+                                .signInWithEmail(_email, _password);
+
+                            // stop loading animation
```

Comment:
Do you remember when I was saying there was something about this that bothered me? I realised what it was I think - I played around with different ways of doing this login process and ended up having a piece of state, called (eg.) authState that the UI observes and the services can update.  It becomes more useful when there are several stages, like in the AuthProvider flows.  Anyway no need to change now imo, just wanted to flag it for discussion at some point.

Reply:
really keen to talk about it nick! I was working on a solution that would work like that and kept running into bugs all week, still couldnt really wrap my head around it so i went with the solution we worked on to get some momentum (productivity wise). Shouldve asked for some help earlier but still learned abit :) 
Lets chat tommorow! Ill be at the meetup, or just anytime that suits

Reply:
Ok brilliant!  Let's get stuck into it tomorrow :-)